### ISSUE TITLE
MacHyperVSupport 0.9.7 PCI Devices Not Working

### ISSUE THREAD

**Issue by PIKACHUIM** [acidanthera/bugtracker#2453]
MacHyperVSupport v0.9.7 not working.
Host System: Windows Server 2025
VM System: macOS 14.7.1
I can't see pci devices passthrougth(DDA) from host
However, MacHyperVSupport v0.9.6 works.
Here is my config:
[config.zip](https://github.com/user-attachments/files/18048127/config.zip)
By the way, I wonder if MacHyperVSupport can support GPU working with CL/Metal.
And will it work with OCLP? 

Thank you!

**Comment by mikebeaton** [comment]
Please [bisect](https://github.com/acidanthera/bugtracker/issues/2436#issuecomment-2436352282) to find the specific commit which introduces the issue.

**Comment by PIKACHUIM** [comment]
Sorry for reply late.
I spent some time ruling out the possibility that this issue could be caused by the opencore version or the hardware itself. Therefore, it was still due to the MacHyperVSupport version that caused the inability to use PCI devices. I tested the builds of various versions from 0.9.6 to 0.9.7, and the following are the results:

> ......(work)
> d9dd44d-work. Boot fast and dda pci devices work.
> c0344f9-not bootable. It shows disk1,4 error, however i do not have this disk.
> a3740c8-not work. Boot very slow and dda pci devices not show in system.
> ......(not work)

Thank you.

**Comment by PIKACHUIM** [comment]
@mikebeaton

**Comment by PIKACHUIM** [comment]
On version-d9dd44d, there are two vgpu show in hackintool instead of one in 0.9.7/0.9.6-release:

![QQ20241222-173056](https://github.com/user-attachments/assets/fc330fd8-6064-454e-8a93-0df2880550d0)

The third pci device appears on this build(d9dd44d)
I just want to drive the second pci device GT720 to get metal support.
And I have already apply OCLP's patch but it seems not work.
Can you help me to make it possible? I am very grateful.

**Comment by mikebeaton** [comment]
@Goldfish64

**Comment by Goldfish64** [comment]
Just for reference, what is the host hardware here?

**Comment by PIKACHUIM** [comment]
Sure, here is the host hardware:
Computer:

Computer Type: ASRock Z690 Phantom Gaming 4
Operating System: Microsoft Windows Server 2025 Datacenter
Operating System Version Upgrade (SP): -
Internet Explorer: 11.1.26100.0
Edge: 131.0.2903.99
DirectX: DirectX 12.0
Computer Name: PC21-W11PC1-CD1
User Name: Pika
Logon Domain: PC21-W11PC1-CD1
Date/Time: 2024-12-25 / 22:13
Motherboard:

Processor Name: i9-13900K 8C+16c, 4300 MHz (43 x 100)
Motherboard Name: ASRock Z690 Phantom Gaming 4 (3 PCI-E x1, 2 PCI-E x16, 3 M.2, 4 DDR4 DIMM, Audio, Video, Gigabit LAN)
Motherboard Chipset: Intel Alder Point-S Z690, Intel Raptor Lake-HX
System Memory: 65104 MB (DDR4 SDRAM)
DIMM1: Gloway International TAC4U3200E18321C 32 GB DDR4-3200 DDR4 SDRAM (18-22-22-42 @ 1600 MHz) (16-20-20-38 @ 1422 MHz)
DIMM3: Gloway International TAC4U3200E18321C 32 GB DDR4-3200 DDR4 SDRAM (18-22-22-42 @ 1600 MHz) (16-20-20-38 @ 1422 MHz)
BIOS Type: AMI (10/01/2024)
Display Devices:

Display Adapter: Intel(R) RaptorLake-S Mobile Graphics Controller (2 GB)
Display Adapter: NVIDIA GeForce RTX 3060 Ti (8 GB)
3D Accelerator: AMD Radeon HD 7750 (Cape Verde)
3D Accelerator: nVIDIA GeForce GT 720
3D Accelerator: nVIDIA GeForce RTX 3060 Ti LHR
Monitor: Generic Monitor (BKY248) [NoDB] (34091) {2024}
Monitor: Generic Monitor (DP) [NoDB] (322821573) {2022}
Multimedia:

Audio Adapter: ATI Radeon HDMI @ AMD Cape Verde/Pitcairn/Curacao/Heathrow/Chelsea/Venus - High Definition Audio Controller
Audio Adapter: nVIDIA GA104 HDMI/DP @ nVIDIA GA104 - High Definition Audio Controller
Audio Adapter: nVIDIA GK208 HDMI/DP @ nVIDIA GK208 - High Definition Audio Controller
Audio Adapter: Realtek ALC897 @ Intel Alder Point-S PCH - HD Audio (Audio, Voice, Speech)
Storage Devices:

IDE Controller: Standard SATA AHCI Controller
Storage Controller: Microsoft VHD Loopback Controller
Storage Controller: Microsoft Storage Spaces Controller
Storage Controller: Standard NVM Express Controller
Storage Controller: Standard NVM Express Controller
Hard Drive: HYV2TBX4 (1907 GB)
Hard Drive: Lenovo SSD SL700 2TB (1863 GB)
Hard Drive: Samsung SSD 970 EVO Plus 2TB (2000 GB, PCI-E 3.0 x4)
Hard Drive SMART Status: OK
Disk Partitions:

C: (NTFS) 256.0 GB (104.4 GB available)
D: (NTFS) 1600.0 GB (848.8 GB available)
F: (NTFS) 1863.0 GB (331.4 GB available)
Total Size: 3719.0 GB (1284.5 GB available)

**Comment by Goldfish64** [comment]
What is your passthrough configuration? Just installed Sonoma on a VM but only see the framebuffer and the passed-in GPU, not the basic render driver.

**Comment by PIKACHUIM** [comment]
I use DDA passthrough configuration, here is a tool: [DDATools.zip](https://github.com/user-attachments/files/18249507/DDATools.zip)
It just uses DDA passthrough by official command from Microsoft.
_**The basic render driver**_ only appears on this build(d9dd44d).

**Comment by PIKACHUIM** [comment]
DDA passthrough just like this:
```powershell
Dismount-VMHostAssignableDevice -LocationPath <locationpath> -Force
Add-VMAssignableDevice -VMName $VmName -LocationPath <locationpath>
```

**Comment by PIKACHUIM** [comment]
> What is your passthrough configuration? Just installed Sonoma on a VM but only see the framebuffer and the passed-in GPU, not the basic render driver.

And I wonder if your GPU passed in will **work with Metal/OpenCL or not**?

**Comment by Goldfish64** [comment]
Tested with Big Sur 11.7.10 and a Quadro K600 that DDA does function, although there are some oddities with backtraces being generated when the cursor moves. The Hyper-V synthetic framebuffer must be disabled with -hvgfxboff boot argument.

Thus far the only GPUs I've been able to get working are Kepler cards on the respective supported versions of macOS. AMD cards do not appear to work at this time. Anything above Monterey for NVIDIA cards would be out of scope of this bugtracker as that would require OCLP.

**Comment by PIKACHUIM** [comment]
> Tested with Big Sur 11.7.10 and a Quadro K600 that DDA does function, although there are some oddities with backtraces being generated when the cursor moves. The Hyper-V synthetic framebuffer must be disabled with -hvgfxboff boot argument.
> 
> Thus far the only GPUs I've been able to get working are Kepler cards on the respective supported versions of macOS. AMD cards do not appear to work at this time. Anything above Monterey for NVIDIA cards would be out of scope of this bugtracker as that would require OCLP.

I have a GT720 and I have installed macOS 11.7.10 but it does not work.
I boot with `-hvgfxboff` argument, and the use MacHyperVSupport v0.9.6.
(I can NOT use MacHyperVSupport v0.9.6 because it does not work at all)
If without `-hvgfxboff`,  I can see the GT720 shows in the pci devices list:
![QQ20241228-224137](https://github.com/user-attachments/assets/70e3e8c8-867e-4e4b-b2e2-1d555a3c2aeb)
If I boot with `-hvgfxboff`, I can NOT see login screen in Hyper-V and vnc can NOT connect.
Here is system report and screenshot from Hackintool:
![QQ20241228-224151](https://github.com/user-attachments/assets/9d41e283-cc85-43ea-92fb-ef1eb219b722)
![QQ20241228-224216](https://github.com/user-attachments/assets/6d34b211-2fc4-428b-a1a8-d51f765d115c)

**Comment by PIKACHUIM** [comment]
Only once I boot with `-hvgfxboff`, I can see the system report and pci devices list only have my GT720.
There is no Hyper-V vGPU. But it still not work, metal shows not support. (I forgot to take a screen shot)

**Comment by PIKACHUIM** [comment]
![QQ20241228-225415](https://github.com/user-attachments/assets/5b09c378-8af4-4f8e-a1c8-1c73d56b1dd5)

**Comment by PIKACHUIM** [comment]
![QQ20241228-225507](https://github.com/user-attachments/assets/3f418b84-99ab-419b-8abe-ba822aa228d4)

**Comment by PIKACHUIM** [comment]
For version 0.9.9, it still not work. Can anyone fix or revert this commit? I will be very grateful.

> Sorry for reply late. I spent some time ruling out the possibility that this issue could be caused by the opencore version or the hardware itself. Therefore, it was still due to the MacHyperVSupport version that caused the inability to use PCI devices. I tested the builds of various versions from 0.9.6 to 0.9.7, and the following are the results:
> 
> > ......(work)
> > d9dd44d-works. Boot fast and dda pci devices work.
> > c0344f9-not bootable. It shows disk1,4 error, however I do not have this disk.
> > a3740c8-not work. Boot very slow and dda pci devices not show in system.
> > ......(not work)
> 
> Thank you.
